src/main/java/duke/list/FunnyList.java

```diff
@@ -0,0 +1,99 @@
+package duke.list;
+
+import java.util.ArrayList;
+
+import duke.exception.DukeException;
+import duke.task.Task;
+
+/**
+ * Represents a list of tasks in the Duke application.
+ * This class manages a list of tasks and provides methods for adding, retrieving,
+ * completing, undoing, and deleting tasks.
+ */
+public class FunnyList {
```

Comment:
I didn't quite like the naming of this class 😞 It may be confusing to the other readers. Perhaps `TaskList` would be a better name?

Reply:
okay i will do something about this thank you!